tests/pocket_test.cpp

```diff
@@ -0,0 +1,832 @@
+            pocket_bucket.insert_item( ketchup );
+            REQUIRE_FALSE( pocket_bucket.empty() );
+
+            THEN( "it can contain more of the same liquid" ) {
+                expect_can_contain( pocket_bucket, ketchup );
+            }
+
+            THEN( "it cannot contain a different liquid" ) {
+                expect_cannot_contain( pocket_bucket, mustard, "can't mix liquid with contained item",
+                                       item_pocket::contain_code::ERR_LIQUID );
+            }
+
+            THEN( "it cannot contain a non-liquid" ) {
+                expect_cannot_contain( pocket_bucket, rock, "can't put non liquid into pocket with liquid",
+                                       item_pocket::contain_code::ERR_LIQUID );
+            }
+        }
+
+        WHEN( "pocket already contains a non-liquid" ) {
+            pocket_bucket.insert_item( rock );
+            REQUIRE_FALSE( pocket_bucket.empty() );
+
+            THEN( "it can contain other non-liquids" ) {
+                expect_can_contain( pocket_bucket, glock );
+            }
+
+            THEN( "it cannot contain a liquid" ) {
+                expect_cannot_contain( pocket_bucket, ketchup, "can't mix liquid with contained item",
+                                       item_pocket::contain_code::ERR_LIQUID );
+                expect_cannot_contain( pocket_bucket, mustard, "can't mix liquid with contained item",
+                                       item_pocket::contain_code::ERR_LIQUID );
+            }
+        }
+    }
+
+    GIVEN( "pocket is not watertight" ) {
+        // Poke a hole in the bucket
+        data_bucket.watertight = false;
+        item_pocket pocket_bucket( &data_bucket );
+
+        THEN( "it cannot contain liquid items" ) {
+            expect_cannot_contain( pocket_bucket, ketchup, "can't contain liquid",
+                                   item_pocket::contain_code::ERR_LIQUID );
+            expect_cannot_contain( pocket_bucket, mustard, "can't contain liquid",
+                                   item_pocket::contain_code::ERR_LIQUID );
+        }
+
+        THEN( "it can still contain non-liquid items" ) {
+            expect_can_contain( pocket_bucket, rock );
+            expect_can_contain( pocket_bucket, glock );
+        }
+    }
+}
+
+// Airtight pockets
+// ----------------
+// To contain gases, a pocket must have the "airtight" attribute set to true.
+//
+// Related JSON fields:
+// "airtight"
+//
+// Functions:
+// item_pocket::airtight
+//
+TEST_CASE( "pockets containing gases", "[pocket][airtight][gas]" )
+{
+    item gas( "test_gas", 0, item::default_charges_tag{} );
+
+    // A potentially airtight container
+    pocket_data data_balloon( item_pocket::pocket_type::CONTAINER );
+
+    // Has capacity for several charges of gas
+    data_balloon.volume_capacity = 4 * gas.volume();
+    data_balloon.max_contains_weight = 4 * gas.weight();
+    // Avoid any length restrictions
```

Comment:
TODO: add exception for gases for length and max_item_volume

Reply:
I included a related test for gases fitting through `max_item_volume` (like soft/liquid items do) but didn't get around to one for length restrictions. Maybe next PR :-)